packages/client/src/pages/common/profile/profile-button.tsx

```diff
@@ -27,10 +31,22 @@ const ProfileButton = (props: any) => {
     setOpen(false);
   };
 
-  // const dispatch = useAppDispatch();
-  // const user = useAppSelector((state) => state.user);
-  // TODO: replace with log out action
+  const [profileModalOpen, setProfileModalOpen]: [
+    boolean,
+    Dispatch<SetStateAction<boolean>>,
+  ] = useState<boolean>(false);
```

Comment:
just fyi, the generic type in `useState` automatically infers the types for the array items (doesn't rly matter tho just a nitty gritty bitty)